services/mail/pod-inbound-mail/src/handlerMta.ts

```diff
@@ -36,6 +36,13 @@ interface MtaMessage {
 
 export async function handleMtaHook (req: Request, res: Response): Promise<void> {
   try {
+    if (config.hookToken !== undefined) {
+      const token = req.headers['x-hook-token']
+      if (token !== config.hookToken) {
+        throw new Error('Invalid hook token')
```

Comment:
Will the 500 response status return In this case?
I think we should return 401 response code in case of incorrect token

Reply:
No, it always retrurn 200 in order not to prevent stalwart from sending emails